File: python/check_if_n_and_its_double_exist.py
from typing import List


class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        elements = set()
        for number in arr:
            if 2 * number in elements or (number % 2 == 0 and number // 2 in elements):
                return True
            elements.add(number)
        return False

File: src/CheckIfNAndItsDoubleExist.java
import java.util.HashSet;
import java.util.Set;

public class CheckIfNAndItsDoubleExist {
    public boolean checkIfExist(int[] array) {
        Set<Integer> elements = new HashSet<>();
        for(int element : array) {
            if (elements.contains(element * 2) || (element % 2 == 0 && elements.contains(element / 2))) {
                return true;
            }
            elements.add(element);
        }
        return false;
    }
}
